src/core/bptree_set.h

```diff
@@ -435,7 +437,24 @@ void BPTree<T, Policy>::ToRank(uint32_t rank, BPTreePath* path) const {
   assert(root_ && rank < count_);
   BPTreeNode* node = root_;
 
+  if (rank + 1 == count_) {
+    // Corner case where we search for the node on the right.
+    while (!node->IsLeaf()) {
+      path->Push(node, node->NumItems());
+      node = node->Child(node->NumItems());
+    }
+    path->Push(node, node->NumItems() - 1);
```

Comment:
with `node->IsLeaf()` that's just 0?

Reply:
nope. Consider a root with 2 items. in that case we won't enter the while loop and want to fetch the second one, index=1